webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/RudderUserDetailsFile.scala

```diff
@@ -554,8 +554,14 @@ object UserFileProcessing {
               // accept both "role" and "roles"
               userRoles(node.attribute("role")) ++ userRoles(node.attribute("roles"))
             ) match {
-              case (Some(name :: Nil), Some(pwd :: Nil), roles) if (name.size > 0) =>
-                Some(ParsedUser(name, pwd, roles)).succeed
+              case (Some(name :: Nil), pwd, roles) if (name.size > 0) =>
+                // password can be optional when an other authentication backend is used.
+                // If the attribute is defined several time, use the first occurrence.
+                val password = pwd match {
+                  case Some(p :: _) => p
+                  case _            => ""
```

Comment:
in "plain text" case, down't this make the account accessible without password?

Reply:
I thought that was what you wanted, i was wondering if this would make an account fallback to nopassword if other authentication fails

Reply:
note that we weren't checking the lengh either before the change. 
I think that we can just set "password" to some random (long enough) string when not defined. In the case of external auth, it will be ignored, and in the case of file based, it will disable login in practice.